windows/devices/uwb/UwbConnector.cxx

```diff
@@ -690,7 +690,7 @@ UwbConnector::HandleNotifications(std::stop_token stopToken)
 
     auto handleDriver = m_notificationHandleDriver;
 
-    while (!stopToken.stop_requested()) {
+    while (!stopToken.stop_requested() && handleDriver.is_valid()) {

```

Comment:
`is_valid()` on a handle will tell you that the handle value is `!= INVALID_HANDLE_VALUE`. So, it's a check that's only valuable after the handle is opened (and this is already done for this handle in [`OpenDriverHandle`](https://github.com/microsoft/nearobject-framework/blob/c2b7c367acbb12749336e2f5de865dc7eb45ed75/windows/devices/util/include/windows/devices/DeviceHandle.hxx#L60), but with an explicit check instead of using `is_valid()`).

I believe the only way to detect the condition mentioned in the description is to handle `ERROR_DRIVER_PROCESS_TERMINATED` directly from the IOCTL, and in that case, invalidate the handle by calling `.reset()` on it (which will cause `is_valid()` to return false) and keep this added check in the PR.

Reply:
Ah I see, I wasn't sure if the driver crash would cause the handle to automatically become invalid or not. I can handle the `ERROR_DRIVER_PROCESS_TERMINATED` error like you said